File: src/main/java/gg/mineral/server/network/packet/play/clientbound/SpawnPositionPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.server.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public record SpawnPositionPacket(int x, int headY, int z) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(x);
        os.writeInt(headY);
        os.writeInt(z);
    }

    @Override
    public byte getId() {
        return 0x05;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/SpawnPositionPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class SpawnPositionPacket(val x: Int, val headY: Int, val z: Int) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) =
        os.writeInt(x, headY, z)

    override val id: Byte
        get() = 0x05
}
